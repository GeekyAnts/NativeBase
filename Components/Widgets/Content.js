/* @flow */
'use strict';

import React from 'react';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import NativeBaseComponent from '../Base/NativeBaseComponent';
import computeProps from '../../Utils/computeProps';


export default class Content extends NativeBaseComponent {

  propTypes: {
        padder : React.PropTypes.bool,
    style : React.PropTypes.object
    }

  prepareRootProps() {

    var type = {
      backgroundColor: 'transparent',
      flex: 1
    }

    var defaultProps = {
      style: type
    }

    return computeProps(this.props, defaultProps);
  }

  render() {
    const contentContainerStyle = this.props.contentContainerStyle || {};
    contentContainerStyle.padding = (this.props.padder) ? this.getTheme().contentPadding : 0;
    return(
      <KeyboardAwareScrollView ref={c => this._scrollview = c} resetScrollToCoords={{x:0,y:0}} {...this.prepareRootProps()} contentContainerStyle={contentContainerStyle}>{this.props.children}</KeyboardAwareScrollView>
    );
  }
}
