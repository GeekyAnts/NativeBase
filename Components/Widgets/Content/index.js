/* @flow */
'use strict';

import React from 'react';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import NativeBaseComponent from '../../Base/NativeBaseComponent';
import computeProps from '../../../Utils/computeProps';


export default class Content extends NativeBaseComponent {

  propTypes: {
      padder: React.PropTypes.bool,
      style : React.PropTypes.object,
      disableKBDismissScroll: React.PropTypes.bool
    }

  prepareRootProps() {

    var type = {
      backgroundColor: 'transparent',
      flex: 1
    }

    var defaultProps = {
      style: type,
      resetScrollToCoords: (this.props.disableKBDismissScroll) ? null : {
        x: 0,
        y: 0
      }
    }

    return computeProps(this.props, defaultProps);
  }

  render() {
    const contentContainerStyle = this.props.contentContainerStyle || {};
    contentContainerStyle.padding = (this.props.padder) ? this.getTheme().contentPadding : 0;
    return(
      <KeyboardAwareScrollView automaticallyAdjustContentInsets={false} ref={(c) => {this._scrollview = c; this._root = c;}} {...this.prepareRootProps()} contentContainerStyle={contentContainerStyle}>{this.props.children}</KeyboardAwareScrollView>
    );
  }
}
