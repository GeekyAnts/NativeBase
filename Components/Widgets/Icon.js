/* @flow */
'use strict';

import React from 'react';
import NativeBaseComponent from '../Base/NativeBaseComponent';
import computeProps from '../../Utils/computeProps';

import Ionicons from 'react-native-vector-icons/Ionicons';
import Entypo from 'react-native-vector-icons/Entypo';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import Foundation from 'react-native-vector-icons/Foundation';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Octicons from 'react-native-vector-icons/Octicons';
import Zocial from 'react-native-vector-icons/Zocial';

export default class IconNB extends NativeBaseComponent {

    propTypes: {
        style : React.PropTypes.object
    }

    contextTypes: {
      theme: React.PropTypes.object
    }

    componentWillMount() {
      switch(this.getTheme().iconFamily) {
        case 'Ionicons':
            this.Icon = Ionicons;
            break;
        case 'Entypo':
            this.Icon = Entypo;
            break;
        case 'FontAwesome':
            this.Icon = FontAwesome;
            break;
        case 'Foundation':
            this.Icon = Foundation;
            break;
        case 'MaterialIcons':
            this.Icon = MaterialIcons;
            break;
        case 'Octicons':
            this.Icon = Octicons;
            break;
        case 'Zocial':
            this.Icon = Zocial;
            break;
        default:
            this.Icon = Ionicons;
      }
    }

    getInitialStyle() {
        return {
            icon: {
                fontSize: this.getTheme().iconFontSize,
                color: this.getContextForegroundColor()
            }
        }
    }
    prepareRootProps() {
        var defaultProps = {
            style: this.getInitialStyle().icon
        };

        return computeProps(this.props, defaultProps);

    }

    render() {
        return(
            <this.Icon {...this.prepareRootProps()}/>
        );
    }
}
