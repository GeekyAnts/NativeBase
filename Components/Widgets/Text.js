/* @flow */
'use strict';

import React from 'react';
import {Text} from 'react-native';
import NativeBaseComponent from '../Base/NativeBaseComponent';
import computeProps from '../../Utils/computeProps';


export default class TextNB extends NativeBaseComponent {

	prepareRootProps() {

		var type = { 
			color: this.getTheme().textColor,
			fontSize: this.getTheme().fontSizeBase,
			lineHeight: this.getTheme().lineHeight
		}

		var defaultProps = {
			style: type
		}	  

		return computeProps(this.props, defaultProps);

	}
	render() {
		return(
			<Text {...this.prepareRootProps()}>{this.props.children}</Text>
		);
	}    

}

