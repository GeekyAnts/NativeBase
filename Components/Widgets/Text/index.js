/* @flow */
'use strict';

import React from 'react';
import {Text} from 'react-native';
import NativeBaseComponent from '../../Base/NativeBaseComponent';
import computeProps from '../../../Utils/computeProps';


export default class TextNB extends NativeBaseComponent {

  propTypes: {
        style : React.PropTypes.object
    }

  prepareRootProps() {

    var type = {
      color: this.getContextForegroundColor(),
      fontSize: this.getTheme().fontSizeBase,
      lineHeight: this.getTheme().lineHeight,
      fontFamily: this.getTheme().fontFamily,
      backgroundColor: this.getTheme().textBgColor
    }

    var defaultProps = {
      style: type
    }

    return computeProps(this.props, defaultProps);

  }
  render() {
    return(
      <Text ref={c => this._root = c} {...this.prepareRootProps()}>{this.props.children}</Text>
    );
  }

}
