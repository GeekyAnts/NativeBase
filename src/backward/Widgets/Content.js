import React, { Component } from 'react';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { connectStyle } from '@shoutem/theme';
import mapPropsToStyleNames from '../../Utils/mapPropsToStyleNames';
import NativeBaseComponent from '../Base/NativeBaseComponent';

class Content extends NativeBaseComponent {
  render() {
    return (
      <KeyboardAwareScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={this.props.contentContainerStyle}
        resetScrollToCoords={(this.props.disableKBDismissScroll) ? null : { x: 0, y: 0 }}
        ref={(c) => { this._scrollview = c; this._root = c; }}
        {...this.props}
      >
        {this.props.children}
      </KeyboardAwareScrollView>
    );
  }
}

Content.propTypes = {
  ...KeyboardAwareScrollView.propTypes,
  style: React.PropTypes.object,
  padder: React.PropTypes.bool,
  disableKBDismissScroll: React.PropTypes.bool,
};

const StyledContent = connectStyle('NativeBase.Content', {}, mapPropsToStyleNames)(Content);

export {
  StyledContent as Content,
};
