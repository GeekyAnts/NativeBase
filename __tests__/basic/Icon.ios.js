import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { Content } from "./../../src/basic/Content";
import { ListItem } from "./../../src/basic/ListItem";
import { Icon } from "./../../src/basic/Icon";
import { Text } from "./../../src/basic/Text";

// Note: test renderer must be required after react-native.

jest.mock("Platform", () => {
  const Platform = require.requireActual("Platform");
  Platform.OS = "ios";
  return Platform;
});

it("renders Icon and Icons with icon family", () => {
  const tree = renderer
    .create(
      <Content padder>
        <ListItem>
          <Icon name="logo-apple" style={{ width: 45, height: 45 }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Ionicons</Text>
          <Icon type="Ionicons" name="home" style={{ color: "#999" }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Entypo</Text>
          <Icon type="Entypo" name="home" style={{ color: "#999" }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Feather</Text>
          <Icon type="Feather" name="home" style={{ color: "#999" }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>FontAwesome</Text>
          <Icon type="FontAwesome" name="home" style={{ color: "#999" }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Foundation</Text>
          <Icon type="Foundation" name="home" style={{ color: "#999" }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Material Icons</Text>
          <Icon type="MaterialIcons" name="home" style={{ color: "#999" }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Material Community Icons</Text>
          <Icon
            type="MaterialCommunityIcons"
            name="home"
            style={{ color: "#999" }}
          />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Octicons</Text>
          <Icon type="Octicons" name="home" style={{ color: "#999" }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Zocial</Text>
          <Icon type="Zocial" name="dropbox" style={{ color: "#999" }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Simple Line Icons</Text>
          <Icon type="SimpleLineIcons" name="home" style={{ color: "#999" }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Evil Icons</Text>
          <Icon type="EvilIcons" name="trophy" style={{ color: "#999" }} />
        </ListItem>
      </Content>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
