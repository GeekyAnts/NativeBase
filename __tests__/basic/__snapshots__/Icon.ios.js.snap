// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Icon and Icons with icon family 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": 10,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  padder={true}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={Object {}}
  viewIsInsideTabBar={false}
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {},
                Object {
                  "height": 45,
                  "width": 45,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "width": 220,
              },
            ]
          }
          uppercase={false}
        >
          Ionicons
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {},
                Object {
                  "color": "#999",
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="Ionicons"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "width": 220,
              },
            ]
          }
          uppercase={false}
        >
          Entypo
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {},
                Object {
                  "color": "#999",
                },
              ],
              Object {
                "fontFamily": "Entypo",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="Entypo"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "width": 220,
              },
            ]
          }
          uppercase={false}
        >
          Feather
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {},
                Object {
                  "color": "#999",
                },
              ],
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="Feather"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "width": 220,
              },
            ]
          }
          uppercase={false}
        >
          FontAwesome
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {},
                Object {
                  "color": "#999",
                },
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="FontAwesome"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "width": 220,
              },
            ]
          }
          uppercase={false}
        >
          Foundation
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {},
                Object {
                  "color": "#999",
                },
              ],
              Object {
                "fontFamily": "fontcustom",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="Foundation"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "width": 220,
              },
            ]
          }
          uppercase={false}
        >
          Material Icons
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {},
                Object {
                  "color": "#999",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="MaterialIcons"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "width": 220,
              },
            ]
          }
          uppercase={false}
        >
          Material Community Icons
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {},
                Object {
                  "color": "#999",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="MaterialCommunityIcons"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "width": 220,
              },
            ]
          }
          uppercase={false}
        >
          Octicons
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {},
                Object {
                  "color": "#999",
                },
              ],
              Object {
                "fontFamily": "Octicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="Octicons"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "width": 220,
              },
            ]
          }
          uppercase={false}
        >
          Zocial
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {},
                Object {
                  "color": "#999",
                },
              ],
              Object {
                "fontFamily": "zocial",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="Zocial"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "width": 220,
              },
            ]
          }
          uppercase={false}
        >
          Simple Line Icons
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {},
                Object {
                  "color": "#999",
                },
              ],
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="SimpleLineIcons"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {},
              Object {
                "width": 220,
              },
            ]
          }
          uppercase={false}
        >
          Evil Icons
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {},
                Object {
                  "color": "#999",
                },
              ],
              Object {
                "fontFamily": "EvilIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="EvilIcons"
        >
          
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
